pkg/microservice/aslan/core/common/service/scmnotify/scmnotify.go

```diff
@@ -408,6 +409,70 @@ func (s *Service) UpdateWebhookCommentForTest(task *task.Task, logger *zap.Sugar
 	return nil
 }
 
+func (s *Service) UpdateWebhookCommentForScanning(task *task.Task, logger *zap.SugaredLogger) (err error) {
+	if task.ScanningArgs.NotificationID == "" {
+		return
+	}
+
+	var notification *models.Notification
+	if notification, err = s.Coll.Find(task.ScanningArgs.NotificationID); err != nil {
+		logger.Errorf("can't find notification by id %s %s", task.TestArgs.NotificationID, err)
+		return err
+	}
+
+	var tasks []*models.NotificationTask
+	var taskExist bool
+	var shouldComment bool
+
+	status := convertTaskStatusToNotificationTaskStatus(task.Status)
+	for _, nTask := range notification.Tasks {
+		if nTask.ID == task.TaskID {
+			shouldComment = nTask.Status != status
+			scmTask := &models.NotificationTask{
+				ProductName:  task.ProductName,
+				TestName:     task.PipelineName,
+				ID:           task.TaskID,
+				ScanningName: task.ScanningArgs.ScanningName,
+				ScanningID:   task.ScanningArgs.ScanningID,
+				Status:       status,
+			}
+
+			tasks = append(tasks, scmTask)
+			taskExist = true
+		} else {
+			tasks = append(tasks, nTask)
+		}
+	}
+
+	if !taskExist {
+		tasks = append(tasks, &models.NotificationTask{
+			ProductName:  task.ProductName,
+			TestName:     task.PipelineName,
+			ID:           task.TaskID,
+			ScanningName: task.ScanningArgs.ScanningName,
+			ScanningID:   task.ScanningArgs.ScanningID,
+			Status:       status,
+		})
+		shouldComment = true
+	}
+
+	if shouldComment {
+		notification.Tasks = tasks
+		if err = s.Client.Comment(notification); err != nil {
+			logger.Errorf("failed to comment %s, %v", notification.ToString(), err)
```

Comment:
Return error?

Reply:
The notification collection has to be inserted for further functionality. So i can't return until the db is updated